a3m/executeOrRunSubProcess.py

```diff
@@ -131,25 +131,24 @@ def launchSubProcess(
 def createAndRunScript(
     text, stdIn="", printing=False, arguments=[], env_updates={}, capture_output=True
 ):
-    # Output the text to a /tmp/ file
-    scriptPath = "/tmp/" + uuid.uuid4().__str__()
-    FILE = os.open(scriptPath, os.O_WRONLY | os.O_CREAT, 0o770)
-    os.write(FILE, text.encode("utf8"))
-    os.close(FILE)
-    cmd = [scriptPath]
-    cmd.extend(arguments)
-
-    # Run it
-    ret = launchSubProcess(
-        cmd,
-        stdIn="",
-        printing=printing,
-        env_updates=env_updates,
-        capture_output=capture_output,
-    )
-
-    # Remove the temp file
-    os.remove(scriptPath)
+    with tempfile.NamedTemporaryFile(
+        dir=tempfile.gettempdir(), encoding="utf-8", mode="wt", delete=False
```

Comment:
> `dir=tempfile.gettempdir()`

What's the benefit of doing this?

Reply:
This will return the Python configured temp directory instead of the hardcoded /tmp which _may_ not be the actual configured temp dir.

Reply:
AFAIK NamedTemporaryFile takes care of that [here](https://github.com/python/cpython/blob/a4460f2eb8b9db46a9bce3c450c8b038038a7c93/Lib/tempfile.py#L557).

Reply:
You are correct - not required to be specified. I have removed this and re-tested.